File: BackendFrontendUpdated/ReactAppQuestionGeneration/myapp/src/service/user_service.js
import httpClient from "../http-common";

const registerUser = (course_id, user) => {
  return httpClient.post(`/user/signup/${course_id}`, user);
};

const authenticateUser = (user) => {
  return httpClient.post("/user/signin", user);
};

export default { registerUser, authenticateUser };

File: EXAMPORTALQUESTIONPAPERPOJO/ExamPortalApplication/src/main/java/com/app/service/UserService.java
package com.app.service;



import com.app.dto.LoginRequestDto;
import com.app.dto.UserResponse;
import com.app.entities.User;

public interface UserService {
   
	public User registerUser(Long course_id,User user);
	public UserResponse authenticateUser(LoginRequestDto dto);
//	public List<User> getAllUsers();
	

}
